lightkube/codecs.py

```diff
@@ -72,17 +74,23 @@ def load_all_yaml(stream: Union[str, TextIO], context: dict = None, template_env
         will be used during templating.
     * **template_env** - `jinja2` template environment to be used for templating. When absent a standard
         environment is used.
+    * **create_resources_for_crds** - If True, a generic resource will be created for every version
+        of every CRD found.  Else, no generic resources will be created.  Default is False
 
     **NOTE**: When using the template functionality (setting the context parameter), the dependency
         module `jinja2` need to be installed.
     """
     if context is not None:
         stream = _template(stream, context=context, template_env=template_env)
-    res = []
+    resources = []
     for obj in yaml.safe_load_all(stream):
         if obj is not None:
-            res.append(from_dict(obj))
-    return res
+            res = from_dict(obj)
+            resources.append(res)
+
+            if create_resources_for_crds is True and isinstance(res, CustomResourceDefinition):
```

Comment:
k8s 1.15 doesn't have `apiextensions_v1` but only [apiextensions_v1beta1](https://gtsystem.github.io/lightkube-models/1.15/resources/apiextensions_v1beta1/#customresourcedefinition). Ideally we should be able to generate models from both versions (v1 and v1beta1). If the schema is the same (I didn't really check) we could just check res.kind == 'CustomResourceDefinition' and res.version in supported versions. This will also remove the import issue above.

Reply:
Removed here, but it is also used in a few other places as mentioned [in this comment](https://github.com/gtsystem/lightkube/pull/34#discussion_r906414114)